### ISSUE TITLE
mesh-base in production

### ISSUE THREAD

**Issue by nherment** [rjrodger/seneca-mesh#22]
My understanding is that seneca-mesh is using a centralized location for knowledge of the other services. That centralized location is used as an address book and allows microservices to communicate in a P2P fashion.

Hence a few questions:

- Is the mesh-base a single point of failure ?
- What is the recovery scenario when the base fails (host down) ?

**Comment by AdrianRossouw** [comment]
it's my understanding that the base only exists because there needs to be a service that comes up first, to allow the others to find each other initially.

once the whole system is up, you should be able to bring it down and put it up again without anything in your system being affected.

@rjrodger is that accurate?

**Comment by mirceaalexandru** [comment]
@AdrianRossouw Yes, from my tests I can say that after other nodes are connected, and if mesh is stopped, then the system still works.

**Comment by manoj150283** [comment]
@mirceaalexandru The system works until and unless there is no change into the system. Lets say we have two services running on node 1 and node 2 and mesh running on base node. Here node 1 and node 2 talks to each other. Now suppose base gets down and we put it up again, system continues to work. But if in case any service node is down or there is a need to restart any service node, in that case we also need to restart the other service nodes.

In short, system continues to work after mesh restart unless and until there is o any change into the system.

**Comment by nherment** [comment]
@manoj150283 this is excellent information. Is there any plan to bring the service into an environment more friendly to flexible infrastructures ?

**Comment by avishnyak** [comment]
@manoj150283 I tracked the issue down to the underlying library rjrodger/sneeze/issues/7

Once that is fixed, the dynamic services starting/stopping will work just fine in the mesh.

**Comment by wzrdtales** [comment]
@manoj150283 in other words, this is not yet production ready. may be time to get our hands on it.

**Comment by rjrodger** [comment]
this underlying issue is now fixed, and the network is resilient to base node restarts
